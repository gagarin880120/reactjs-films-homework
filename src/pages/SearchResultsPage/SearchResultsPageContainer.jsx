import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import SearchResultsPage from './SearchResultsPage';
import { resultsSelector, isLoadedSelector } from '../../redux/selectors';
import { getMovies } from '../../redux/actions';
import getURL from '../../utils/utils';

export function SearchResultsPageContainer({ onTrending, results, isLoaded }) {
  useEffect(() => {
    onTrending();
  }, []);

  return (
    <SearchResultsPage
      results={results}
      isLoaded={isLoaded}
    />
  );
}

export const mapStateToProps = (state) => ({
  results: resultsSelector(state),
  isLoaded: isLoadedSelector(state),
})

export const mapDispatchToProps = (dispatch) => ({
  onTrending() {
    dispatch(getMovies(getURL('trending'), 1));
  },
});

SearchResultsPageContainer.propTypes = {
  onTrending: PropTypes.func,
  results: PropTypes.instanceOf(Array),
  isLoaded: PropTypes.bool,
};

SearchResultsPageContainer.defaultProps = {
  onTrending: null,
  results: [],
  isLoaded: false,
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchResultsPageContainer);
