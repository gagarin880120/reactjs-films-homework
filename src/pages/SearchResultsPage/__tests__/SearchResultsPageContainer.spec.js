import React from 'react';
import { SearchResultsPageContainer, mapStateToProps } from '../SearchResultsPageContainer';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('SearchResultsPageContainer', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<SearchResultsPageContainer />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  });


  describe('mapStateToProps', () => {
    test('should return the right value', () => {

      const initialState = {
        areMoviesLoaded: true,
        results: [{title:"Star Wars", id:15, genre_ids:[28]}],
        genres: [{id: 28, name: 'Action'}],
      };

      expect(mapStateToProps(initialState).results).toEqual([
        {title:"Star Wars", id:15, genre_ids:[28], genres: 'Action'}
      ]);
      expect(mapStateToProps(initialState).areMoviesLoaded).toEqual(true);
    });
  });

});
