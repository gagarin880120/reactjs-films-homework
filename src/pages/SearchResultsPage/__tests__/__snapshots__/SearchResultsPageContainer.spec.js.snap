// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsPageContainer component should render without crashing 1`] = `
<SearchResultsPage
  areMoviesLoaded={false}
  results={Array []}
/>
`;
